tests/baselines/reference/assignmentCompatWithConstructSignatures5.errors.txt

```diff
@@ -1,14 +1,20 @@
 tests/cases/conformance/types/typeRelationships/assignmentCompatibility/assignmentCompatWithConstructSignatures5.ts(40,1): error TS2322: Type 'new <T>(x: T) => void' is not assignable to type 'new <T>(x: T) => T'.
-  Type 'void' is not assignable to type 'T'.
+  Types of constructor signature are incompatible.
+    Type 'new <T>(x: T) => void' is not assignable to type 'new <T>(x: T) => T'.
+      Type 'void' is not assignable to type 'T'.
 tests/cases/conformance/types/typeRelationships/assignmentCompatibility/assignmentCompatWithConstructSignatures5.ts(55,1): error TS2322: Type 'new <T>(x: { a: T; b: T; }) => T[]' is not assignable to type 'new <U, V>(x: { a: U; b: V; }) => U[]'.
-  Type '(U | V)[]' is not assignable to type 'U[]'.
-    Type 'U | V' is not assignable to type 'U'.
-      Type 'V' is not assignable to type 'U'.
+  Types of constructor signature are incompatible.
+    Type 'new <T>(x: { a: T; b: T; }) => T[]' is not assignable to type 'new <U, V>(x: { a: U; b: V; }) => U[]'.

```

Comment:
This ends up repeating the last line in a lot of places which is definitely a non-optimal experience. I think the problem is that this should only be done when the elaboration wouldn't have otherwise taken place.

Reply:
Do you mean `Type 'new <T>(x: { a: T; b: T; }) => T[]' is not assignable to type 'new <U, V>(x: { a: U; b: V; }) => U[]'.` Is repeated in different place?
And this message should not reported if it is not comparing to a constructor like `let FooConstructor: { new(): Foo } = Foo;`?  Can you elaborate more? Sorry English is not my native language.

Reply:
@DanielRosenwasser , could you elaborate more?